xml/ha_qdevice-qnetd.xml

```diff
@@ -0,0 +1,508 @@
+         jump to step 3?</remark>
+         <step>
+            <para>
+               On &node1;, initialize your cluster:
+            </para>
+            <screen>&prompt.root;<command>crm</command> cluster init -y</screen>
+         </step>
+         <step>
+            <para>
+               On &node2;, join the cluster:
+            </para>
+            <screen>&prompt.root;<command>crm</command> cluster join -c &node1; -y</screen>
+         </step>
+         <step>
+            <para>
+               On &node1; and &node2;, bootstrap the <literal>qdevice</literal> stage.
+               Usually, in most cases the default settings are fine.
+               Provide at least <option>--qnetd-hostname</option> and the hostname or
+               IP address of the &qnet; server (&node3; in our case):
+            </para>
+            <screen>&prompt.root;<command>crm</command> cluster init qdevice --qnetd-hostname=&node3;</screen>
+            <para>
+               If you want to change the default settings, get a list of all
+               possible options with the command <command
+                  >crm cluster init qdevice --help</command>.
+               All options related to &qdevice; start with
+               <option>--qdevice-<replaceable>NAME</replaceable></option>.
+            </para>
+         </step>
+      </procedure>
+      <para>
+         If you have used the default settings, it creates a &qdevice; with TLS enabled
+         and using the FFSplit algorithm.
+      </para>
+   </sect1>
+
+   <sect1 xml:id="sec-ha-qdevice-heuristic">
+      <title>Setting up a &qdevice; with Heuristics</title>
+      <para>
+         If you need additional control over how votes are determined, use heuristics.
+         Heuristics are a set of commands which are executed in parallel.
+      </para>
+      <para>
+         For this purpose, the command <command>crm cluster init qevice</command>
+         provides the option <option>--qdevice-heuristics</option>. You can
+         pass one or more commands (separated by semicolon) with absolute paths.
+      </para>
+      <para>
+         For example, if your own command for heuristic check is located at
+         <filename>/usr/sbin/my-script.sh</filename> you could run it on
+         one of your cluster nodes as follows:
+      </para>
+      <screen>&prompt.root;<command>crm</command
+> cluster init qdevice --qdevice-hostname=&node3; \
+     --qdevice-heuristics=/usr/sbin/my-script.sh</screen>
+      <para>
+         The command can be written in any language such as Shell, Python, Ruby etc.
+         It returns 0 (zero) if succeded, otherwise it returns an error code.
+      </para>
+      <para>
+         If you need more than one command, you can do so as well. In this case, only
+         when all commands finish successfully (return code is zero), the heuristics
+         has passed.
+      </para>
+   </sect1>
+
```

Comment:
One important thing is the `qdevice-heuristics-mode` option, for me at least because I faced it when I was doing the test in openQA.
From my point of view, if you want to monitor something specific with heuristics, most of the time you want a regular check and not only one check when the cluster stack starts. That's why we are using the option `--qdevice-heuristics-mode=on`, if I'm not wrong, heuristics is checking script return code every 30 seconds by default with this option.
Of course I admit we can let the customer refers to the man page so it is just an idea. However I suppose the purpose is not to document all the things which are in the man but only document the basics.

Reply:
> From my point of view, if you want to monitor something specific with heuristics, most of the time you want a regular check and not only one check when the cluster stack starts. That's why we are using the option `--qdevice-heuristics-mode=on`, 

Yes, if one would like to see what the instant result of the heuristics check could be. Technically a check is only meaningful when membership changes and quorum needs to be updated.  so qdevice-heuristics-mode=sync might be confusing but is technically sufficient.

But I don't mind to add qdevice-heuristics-mode=on for the demonstration :-)

> if I'm not wrong, heuristics is checking script return code every 30 seconds by default with this option.

That's true.

Reply:
@juadk thanks for your help. That's a good tip! :+1: I've integrated it into the crm example and added the following additional sentence:

    The <option>--qdevice-heuristics-mode=on</option> option lets the heuristics
    commands run regularily.